core/drivers/imx_sc_api.c

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ * Copyright 2017-2021 NXP
+ */
+#include <drivers/imx_mu.h>
+#include <drivers/imx_sc_api.h>
+#include <imx-regs.h>
+#include <initcall.h>
+#include <kernel/mutex.h>
+#include <mm/core_memprot.h>
+#include <tee_api_types.h>
+#include <trace.h>
+
+#define RNG_INIT_RETRY 100
+
+#define SC_RPC_VERSION 1
+#define SC_RPC_MAX_MSG 8
+
+/* Macros to fill struct sc_rpc_msg data field */
+#define RPC_U32(mesg, idx) ((mesg)->data.u32[(idx)])
+#define RPC_U16(mesg, idx) ((mesg)->data.u16[(idx)])
+#define RPC_U8(mesg, idx)  ((mesg)->data.u8[(idx)])
+
+/* Defines for struct sc_rpc_msg svc field */
+#define SC_RPC_SVC_PM	2
+#define SC_RPC_SVC_RM	3
+#define SC_RPC_SVC_SECO 9
+
+/* Define for PM function calls */
+enum sc_pm_func {
+	SC_PM_FUNC_SET_RESOURCE_POWER_MODE = 3
+};
+
+/* Defines for RM function calls */
+enum sc_rm_func {
+	SC_RM_FUNC_GET_PARTITION = 5,
+	SC_RM_FUNC_ASSIGN_RESOURCE = 8
+};
+
+/* Define for SECO function calls */
+enum sc_seco_func {
+	SC_SECO_FUNC_START_RNG = 22
+};
+
+/* Internal SCFW API error codes */
+enum sc_error {
+	SC_ERR_NONE = 0,	/* Success */
+	SC_ERR_VERSION,		/* Incompatible API version */
+	SC_ERR_CONFIG,		/* Configuration error */
+	SC_ERR_PARM,		/* Bad parameter */
+	SC_ERR_NOACCESS,	/* Permission error (no access) */
+	SC_ERR_LOCKED,		/* Permission error (locked) */
+	SC_ERR_UNAVAILABLE,	/* Unavailable (out of resources) */
+	SC_ERR_NOTFOUND,	/* Not found */
+	SC_ERR_NOPOWER,		/* No power */
+	SC_ERR_IPC,		/* Generic IPC error */
+	SC_ERR_BUSY,		/* Resource is currently busy/active */
+	SC_ERR_FAIL,		/* General I/O failure */
+	SC_ERR_LAST
+};
+
+/* RNG SECO states */
+enum sc_seco_rng_status {
+	SC_SECO_RNG_STAT_UNAVAILABLE = 0,
+	SC_SECO_RNG_STAT_INPROGRESS,
+	SC_SECO_RNG_STAT_READY
+};
+
+/* Resources IDs */
+enum sc_resource {
+	SC_RES_CAAM_JR1 = 500,
+	SC_RES_CAAM_JR2,
+	SC_RES_CAAM_JR3,
+	SC_RES_CAAM_JR1_OUT = 514,
+	SC_RES_CAAM_JR2_OUT,
+	SC_RES_CAAM_JR3_OUT,
+	SC_RES_CAAM_JR0 = 519,
+	SC_RES_CAAM_JR0_OUT,
+	SC_RES_LAST = 546
+};
+
+/* Power modes */
+enum sc_power_mode {
+	SC_PM_PW_MODE_OFF = 0,
+	SC_PM_PW_MODE_STBY,
+	SC_PM_PW_MODE_LP,
+	SC_PM_PW_MODE_ON
+};
+
+/* RPC message header */
+struct sc_rpc_msg_header {
+	uint8_t version;	/* SC RPC version */
+	uint8_t size;		/* Size of the message */
+	uint8_t svc;		/* Type of service (RM, PM, ...)*/
```

Comment:
space char before `*/`
suggestion: `/* Type of service: one of SC_RPC_SVC_* */`

Reply:
Ok